### ISSUE TITLE
formatMethodMap[type] is not a function in vision lib

### ISSUE THREAD

**Issue by uliporeggia** [GoogleCloudPlatform/gcloud-node#1333]
Just got this error calling `vision.detectText`

```
Debug: internal, implementation, error 
    TypeError: Uncaught error: formatMethodMap[type] is not a function
    at /home/uli/Projetos/projeto-a/server/node_modules/gcloud/lib/vision/index.js:345:39
    at Array.map (native)
    at /home/uli/Projetos/projeto-a/server/node_modules/gcloud/lib/vision/index.js:388:18
    at Array.map (native)
    at /home/uli/Projetos/projeto-a/server/node_modules/gcloud/lib/vision/index.js:377:10
    at /home/uli/Projetos/projeto-a/server/node_modules/gcloud/lib/vision/index.js:158:5
    at Object.handleResp (/home/uli/Projetos/projeto-a/server/node_modules/gcloud/lib/common/util.js:126:3)
    at /home/uli/Projetos/projeto-a/server/node_modules/gcloud/lib/common/util.js:446:12
    at Request.onResponse [as _callback] (/home/uli/Projetos/projeto-a/server/node_modules/retry-request/index.js:117:7)
    at Request.self.callback (/home/uli/Projetos/projeto-a/server/node_modules/gcloud/node_modules/request/request.js:200:22)
```

I am using node v4.3.2 with hapi framework

**Comment by stephenplusplus** [comment]
Could you paste more of your code that triggers this?

**Comment by stephenplusplus** [comment]
I have a feeling it's an annotation error, possibly from a malformed image input type. We won't be able to know for sure unless you go into the node_modules/gcloud directory and edit the lib/vision/index.js file, adding a console.log(resp) around here: https://github.com/GoogleCloudPlatform/gcloud-node/blob/master/lib/vision/index.js#L349

I'll work on a PR asap that handles the type of situation I think you're running into.

**Comment by uliporeggia** [comment]
Yes, of course.
Here is the log
```
{ responses: [ { error: [Object] } ] }
[ { error: 
     { code: 3,
       message: 'image-annotator::Malformed request.: Image processing error!' } } ]
```
Right that, malformed image type!

**Comment by uliporeggia** [comment]
But this is strange because I am not making any manipulation on the image, I am just passing directly from hapi parser to vision API

```javascript
    server.route({
        method : 'POST',
        path   : '/notas',
        config : {
            payload : {
                output   : 'file',
                parse    : true,
                allow    : 'multipart/form-data',
                uploads  : __dirname + '/uploads/',
                maxBytes : 1048576 * 10 //10 MB
            },
            handler : NotasRoutes.create
        }
    });
```

and then

```javascript
var gcloud = require('gcloud');

gcloud = gcloud({
    projectId   : 'projeto-a',
    keyFilename : './credential.json'
});

module.exports.create = function (req, reply) {

    var vision = gcloud.vision();

    var data = req.payload;

    if (data.file) {
        vision.detectText(data.file.path, { verbose: true }, function (err, text, apiResponse) {

            if (err)
                throw err;

            reply({ text: text, apiResponse: apiResponse });

        });
    } else {
        reply({ erro : 'arquivo não encontrado' });
    }
}
```

**Comment by stephenplusplus** [comment]
I put together a PR that will make catching the error easier (#1334) (thanks for bringing this to our attention!)

As far as why the image is malformed, I can't be sure without testing the failing image and observing the steps that it goes through in your set up. Is it possible to put together a barebones repo so I can try locally?

**Comment by uliporeggia** [comment]
This is great! I thank you for answering my issue.
About the image, I think there is no needed more help, there must be some simple thing in my side.

**Comment by JustinBeckwith** [comment]
Yar.  I'm running into the same problem all of the sudden with cloudcats.  Who knows which random reddit image is causing this.  The problem is - I can't seem to find a way to work around the unhandled exception :(

```
error: uncaughtException: formatMethodMap[type] is not a function date=Wed Jun 01 2016 22:56:23 GMT-0700 (PDT), pid=75156, uid=306440, gid=5000, cwd=/Users/beckwith/Code/next/cloudcats/worker, execPath=/usr/local/bin/node, version=v4.2.2, argv=[/usr/local/bin/node, /Users/beckwith/Code/next/cloudcats/worker/server.js], rss=383635456, heapTotal=323568128, heapUsed=278358432, loadavg=[2.63623046875, 2.8525390625, 2.701171875], uptime=829523, trace=[column=39, file=/Users/beckwith/Code/next/cloudcats/worker/node_modules/gcloud/lib/vision/index.js, function=null, line=326, method=null, native=false, column=null, file=null, function=Array.map, line=null, method=map, native=true, column=18, file=/Users/beckwith/Code/next/cloudcats/worker/node_modules/gcloud/lib/vision/index.js, function=null, line=369, method=null, native=false, column=null, file=null, function=Array.map, line=null, method=map, native=true, column=10, file=/Users/beckwith/Code/next/cloudcats/worker/node_modules/gcloud/lib/vision/index.js, function=null, line=358, method=null, native=false, column=5, file=/Users/beckwith/Code/next/cloudcats/worker/node_modules/gcloud/lib/vision/index.js, function=null, line=157, method=null, native=false, column=3, file=/Users/beckwith/Code/next/cloudcats/worker/node_modules/gcloud/lib/common/util.js, function=Object.handleResp, line=126, method=handleResp, native=false, column=12, file=/Users/beckwith/Code/next/cloudcats/worker/node_modules/gcloud/lib/common/util.js, function=null, line=446, method=null, native=false, column=7, file=/Users/beckwith/Code/next/cloudcats/worker/node_modules/gcloud/node_modules/retry-request/index.js, function=Request.onResponse [as _callback], line=117, method=onResponse [as _callback], native=false, column=22, file=/Users/beckwith/Code/next/cloudcats/worker/node_modules/request/request.js, function=Request.self.callback, line=199, method=self.callback, native=false], stack=[TypeError: formatMethodMap[type] is not a function,     at /Users/beckwith/Code/next/cloudcats/worker/node_modules/gcloud/lib/vision/index.js:326:39,     at Array.map (native),     at /Users/beckwith/Code/next/cloudcats/worker/node_modules/gcloud/lib/vision/index.js:369:18,     at Array.map (native),     at /Users/beckwith/Code/next/cloudcats/worker/node_modules/gcloud/lib/vision/index.js:358:10,     at /Users/beckwith/Code/next/cloudcats/worker/node_modules/gcloud/lib/vision/index.js:157:5,     at Object.handleResp (/Users/beckwith/Code/next/cloudcats/worker/node_modules/gcloud/lib/common/util.js:126:3),     at /Users/beckwith/Code/next/cloudcats/worker/node_modules/gcloud/lib/common/util.js:446:12,     at Request.onResponse [as _callback] (/Users/beckwith/Code/next/cloudcats/worker/node_modules/gcloud/node_modules/retry-request/index.js:117:7),     at Request.self.callback (/Users/beckwith/Code/next/cloudcats/worker/node_modules/request/request.js:199:22)]
```

**Comment by stephenplusplus** [comment]
Try out the PR (#1334) branch:

```js
$ npm install --save stephenplusplus/gcloud-node#spp--1333
```

Where you were previously expecting results, say `detections`, there will be a `detections.errors` array. If all goes well, we can merge and release soon.

**Comment by stephenplusplus** [comment]
Released in 0.35.0.
